lib/adapters/definition-adapter.ts

```diff
@@ -46,14 +46,14 @@ export default class DefinitionAdapter {
     const definitionLocations = DefinitionAdapter.normalizeLocations(
       await connection.gotoDefinition(documentPositionParams)
     )
-    if (definitionLocations == null || definitionLocations.length === 0) {
+    if (definitionLocations === null || definitionLocations.length === 0) {
       return null
     }
 
     let queryRange
     if (serverCapabilities.documentHighlightProvider) {
       const highlights = await connection.documentHighlight(documentPositionParams)
-      if (highlights != null && highlights.length > 0) {
+      if (highlights.length > 0) {
```

Comment:
Same here, are we sure connection always returns something? I know with tsserver that is not the case, the official types lie a lot about `null`.